assets/performanceprofile/scripts/set-rps-mask.sh

```diff
@@ -33,6 +38,8 @@ function find_dev_dir {
 [ -d "${dev_dir}" ] || { sleep 5; find_dev_dir; }  # search failed, wait a little and try again
 [ -d "${dev_dir}" ] || { echo "${dev_dir}" directory not found >&2 ; exit 0; } # the interface disappeared, not an error
 
-for i in "${dev_dir}"/queues/rx-*/rps_cpus; do
-  echo "${mask}" > "${i}"
+set-rps-mask
+# check that all queues were setup correctly
+while [[ $(cat /sys/devices/virtual/net/*/queues/*/rps_cpus | sort -u | wc -l) != "1" ]]; do
```

Comment:
Isn't the main issue with this approach that there is a race condition here?

Reply:
Theoretically yes, it's not perfect but it will decrease the problem rate significantly and make the CI much less flaky. The CI is falling most of the time because of the test that checks all queues are properly set. we can have this change til a proper solution will be found.